expat/lib/xmlparse.c

```diff
@@ -6295,6 +6295,30 @@ poolStoreString(STRING_POOL *pool, const ENCODING *enc,
   return pool->start;
 }
 
+static size_t
+poolBytesToAllocateFor(int blockSize)
+{
+  /* Unprotected math would be:
+  ** return offsetof(BLOCK, s) + blockSize * sizeof(XML_Char);
+  **
+  ** Detect overflow, avoiding _signed_ overflow undefined behavior
+  ** For a + b * c we check b * c in isolation first, so that addition of a
+  ** on top has no chance of making us accept a small non-negative number
+  */
+  const int stretchedBlockSize = (int)((unsigned)blockSize * sizeof(XML_Char));
```

Comment:
I force-pushed a remake of that commit with your feedback applied now.
Does it look better now?


Out of curiosity, I wonder how the following "product smaller than factor?" approach compares to the `INT_MAX` approach:
```
if (blockSize <= 0)
  return 0;

const int stretchedBlockSize = (int)(blockSize * sizeof(XML_Char));
if ((stretchedBlockSize < blockSize) || stretchedBlockSize < sizeof(XML_Char))
  return 0;
```
Would it work as well?